pkg/tools/etcd_tools.go

```diff
@@ -418,20 +418,24 @@ func (h *EtcdHelper) AtomicUpdate(key string, ptrToType runtime.Object, ignoreNo
 	}
 }
 
-func checkEtcd(host string) error {
+// Performs a version check against the provided Etcd server, returning a triplet
+// of the release version, internal version, and error (if any).
+func CheckEtcd(host string) (string, string, error) {
 	response, err := http.Get(host + "/version")
 	if err != nil {
-		return err
+		return "", "", err
 	}
 	defer response.Body.Close()
 	body, err := ioutil.ReadAll(response.Body)
 	if err != nil {
-		return err
+		return "", "", err
 	}
-	if !strings.HasPrefix(string(body), "etcd") {
-		return fmt.Errorf("unknown server: %s", string(body))
+
+	var dat map[string]interface{}
+	if err := json.Unmarshal(body, &dat); err != nil {
```

Comment:
Shouldn't be a concern with the current usage of this code (see NewEtcdClientStartServerIfNecessary below). But, could be an issue if other start relying on this code for a true version check.